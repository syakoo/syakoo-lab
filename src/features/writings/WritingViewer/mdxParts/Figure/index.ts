import { Figure as FigureComponent, FigureCaption } from "./Figure";

type Figure = typeof FigureComponent & {
  Caption: typeof FigureCaption;
};

export const Figure = FigureComponent as Figure;
Figure.Caption = FigureCaption;
Figure.Caption.displayName = "Figure.Caption";
