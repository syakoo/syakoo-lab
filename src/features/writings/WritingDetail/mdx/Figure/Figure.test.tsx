import { testStories } from "@/test/testStories";

import * as stories from "./Figure.stories";

describe("Figure", () => {
  testStories(stories);
});
