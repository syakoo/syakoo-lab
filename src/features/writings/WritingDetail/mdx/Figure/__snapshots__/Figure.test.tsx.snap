// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Figure renders NoCaption 1`] = `
<body>
  <div>
    <figure
      class="Figure_figureStyle__4cqq2c0"
    >
      <img
        alt=""
        src="https://placehold.jp/3063D4/ffffff/150x200.png"
      />
    </figure>
  </div>
</body>
`;

exports[`Figure renders Sample 1`] = `
<body>
  <div>
    <figure
      class="Figure_figureStyle__4cqq2c0"
    >
      <img
        alt=""
        src="https://placehold.jp/3063D4/ffffff/400x300.png"
      />
      <figcaption
        class="Figure_figureCaptionStyle__4cqq2c1"
      >
        sample caption
      </figcaption>
    </figure>
  </div>
</body>
`;
